source/code/util/xml/XMLReader.cpp

```diff
@@ -490,19 +531,19 @@ Utf8String XMLReader::_ReduceAttrValue(/*Utf8String& pInOut,*/ Utf8String::Char
         }
 
         // If this was a reference, do the replace
-        if (*m_CharStartPos == c_Ampersand)
+        if (*m_CharStartPos == u8_Ampersand)
         {
-            char c = c_NullChar;
+            char c = u8_NullChar;
             
-            m_CharStartPos++;
+            *m_CharStartPos++;
```

Comment:
Similar change as above w.r.t incrementing+dereferencing an iterator instead of just incrementing.